#Ansible Module Name - ECS
#Test Case : verify that playbook is throwing when User add instance to Security Group With State- Absent and SG_Action- Join
#Ansible Playbook created by - Jogesh N
---
- name: Add instance to Security Group With State- Absent and SG_Action- Join
  hosts: localhost
  connection: local
  gather_facts: False

  vars_files:
    - ../vars/test-environment.yml
    - ../vars/general-parameters.yml
    - ../../../vars/alicloud-creds.yml
  
  tasks:
   - block:
     - name: Add instance to Security Group With State- Absent and SG_Action- Join
       ecs:
        acs_access_key: '{{ aliyun_acs_access_key }}'
        acs_secret_access_key: '{{ aliyun_acs_secret_access_key }}'
        region: '{{ region }}'
        zone: '{{ zone }}'
        group_id: '{{ sgGroupId }}' # VALID Security Group
        instance_ids: '{{ ecsInstanceId }}'# VALID ECS Instance Id
        state: terminate
    #   state: '{{ sgJoin }}'
        sg_action: '{{ sgJoin }}'
       register: result
     - debug: var=result
   
     rescue:
      - fail:
          msg: "The fail block is added just to handle exception. This block won't be executed ever."
        when: no

     always:
      - assert:  
         that:
            - "result.changed ==  false"
            -  "'value of status must be one of: present,running,stopped,restarted,absent,getinfo,getstatus, got: terminate' in result.msg"
         msg: 'Test pass PASSED !!!
               This is a NEGATIVE playbook !!!
               FAILURE - EXPECTED
               REASON: {{ result}}'